projects/storefrontapp-e2e/src/suites/added-to-cart-modal.e2e-spec.ts

```diff
@@ -0,0 +1,122 @@
+import { ProductDetailsPage } from '../page-objects/product-details.po';
+import { AddedToCartModal } from '../page-objects/cmslib/added-to-cart-modal.po';
+
+describe('Added to cart modal', () => {
+  let productDetails: ProductDetailsPage;
+  let addedToCartModal: AddedToCartModal;
+  const productId = '3595723';
+  const productId2 = '3325048';
+
+  beforeEach(async () => {
+    productDetails = new ProductDetailsPage();
+    addedToCartModal = new AddedToCartModal();
+  });
+
+  it('basic modal behavior', async () => {
+    await productDetails.navigateTo(productId);
+    await productDetails.itemCounterUpButton.click();
+    await productDetails.addToCart();
+    await addedToCartModal.waitForReady();
+    // correct number of items are added to cart
+    expect(await addedToCartModal.modalTitle.getText()).toEqual(
+      '2 item(s) added to your cart'
+    );
+    // good product is added to cart
+    expect(await addedToCartModal.itemName.getText()).toEqual(
+      await productDetails.productTitle.getText()
+    );
+    // quantity is set correctly
+    expect(await addedToCartModal.itemQuantity.getText()).toEqual('2');
+    expect(await addedToCartModal.totalCount.getText()).toContain('2 items');
+    // actin buttons links correctly
+    expect(
+      await addedToCartModal.viewCartButton.getAttribute('href')
+    ).toContain('/cart');
+    expect(
+      await addedToCartModal.goToCheckoutButton.getAttribute('href')
+    ).toContain('/checkout');
+    // closing modal works
+    await addedToCartModal.closeButton.click();
+    expect(addedToCartModal.modal.isPresent()).toBe(false);
+  });
+
+  it('adding same product twice to cart', async () => {
+    // add 2 items to cart
+    await productDetails.navigateTo(productId);
+    await productDetails.itemCounterUpButton.click();
+    await productDetails.addToCart();
+    await addedToCartModal.waitForReady();
+    await addedToCartModal.closeButton.click();
+
+    // add next 3 items of the same product to cart
+    await productDetails.itemCounterUpButton.click();
+    await productDetails.addToCart();
+    await addedToCartModal.waitForReady();
+
+    expect(await addedToCartModal.modalTitle.getText()).toEqual(
+      '3 item(s) added to your cart'
+    );
+    // quantity is correctly updated
+    expect(await addedToCartModal.itemQuantity.getText()).toEqual('5');
+    expect(await addedToCartModal.totalCount.getText()).toContain('5 items');
+  });
+
+  // enable this test after fixing cart resetting on each full page load
```

Comment:
Could you add `// TODO`. It easier later to find all the things we would have to fix.